dspace-api/src/main/java/org/dspace/xmlworkflow/storedcomponents/XmlWorkflowItem.java

```diff
@@ -134,4 +139,49 @@ public void setPublishedBefore(boolean b) {
         this.publishedBefore = b;
     }
 
+    @Override
+    public void update() throws SQLException, AuthorizeException {
+
+        Context context = null;
+        try {
+            context = new Context();
+            WorkflowServiceFactory.getInstance().getWorkflowItemService().update(context, this);
+        } finally {
+            if (context != null && context.isValid()) {
+                context.abort();
+            }
+        }
+    }
+
+    @Override
+    public int getState() {
+        // FIXME
+        return 0;
```

Comment:
Please explain this "FIXME" further.  How should this be fixed/improved?

Reply:
done https://github.com/DSpace/DSpace/pull/2312/files#diff-1a554ce535923e7c211241af6f7385c8R158